common/timeout.h

```diff
@@ -52,8 +52,11 @@ class Timeout {
             m_expiration = x;
         return *this;
     }
+    using us = std::chrono::microseconds;
+    template<class Rep, class Period = std::ratio<1>>
+    Timeout(const std::chrono::duration<Rep, Period>& x) :
```

Comment:
since duration is a signed value, Timeout to duration define might be not quite well-defined.
Timeout from a negative duration should consider as nearly infinite timeout or as 0 timeout might not definite here.